### ISSUE TITLE
Import can be added automatically or through action

### ISSUE THREAD

**Issue by RodgeFu** [microsoft/typespec#4862]
### Clear and concise description of the problem

when user uses something from external library, the corresponding import will be added automatically or through action just like what we have when editing typescript.

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [docs](https://typespec.io/docs/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by RodgeFu** [comment]
@timotheeguerin , for this issue, the only solution i can think about is to compiler all the imported typespec libraries and cache their symbols so that we can have enough information to provide suggestions when needed. Do you have any better ideas? Or are you aware of anything that compiler may support that will help on this scenario so that we may should hold on fixing this issue now? thanks.

**Comment by timotheeguerin** [comment]
Yeah I would hold on solving that one, this is most likely a major compiler change
